lib/Runtime/Language/JavascriptStackWalker.cpp

```diff
@@ -298,147 +298,211 @@ namespace Js
-                )
+            if (this->interpreterFrame)
             {
-                uint32 offset = this->interpreterFrame->GetReader()->GetCurrentOffset();
-                if (offset == 0)
+                if (this->TryGetByteCodeOffsetFromInterpreterFrame(offset))
                 {
-                    // This will be the case when we are broken on the debugger on very first statement (due to async break).
-                    // Or the interpreter loop can throw OOS on entrance before executing bytecode.
                     return offset;
                 }
-                else
-                {
-                    // Note : For many cases, we move the m_currentLocation of ByteCodeReader already to next available opcode.
-                    // This could create problem in binding the exception to proper line offset.
-                    // Reducing by 1 will make sure the current offset falls under, current executing opcode.
-                    return offset - 1;
-                }
             }
 
 #if ENABLE_NATIVE_CODEGEN
-            DWORD_PTR pCodeAddr;
-            uint loopNum = LoopHeader::NoLoop;
-            if (this->lastInternalFrameInfo.codeAddress != nullptr)
+            if (TryGetByteCodeOffsetFromNativeFrame(offset))
             {
-                if (this->lastInternalFrameInfo.loopBodyFrameType == InternalFrameType_LoopBody)
-                {
-                    AnalysisAssert(this->interpreterFrame);
-                    loopNum = this->interpreterFrame->GetCurrentLoopNum();
-                    Assert(loopNum != LoopHeader::NoLoop);
-                }
-
-                pCodeAddr = (DWORD_PTR)this->lastInternalFrameInfo.codeAddress;
-            }
-            else
-            {
-                if (this->IsCurrentPhysicalFrameForLoopBody())
-                {
-                    // Internal frame but codeAddress on lastInternalFrameInfo not set. We must be in an inlined frame in the loop body.
-                    Assert(this->tempInterpreterFrame);
-                    loopNum = this->tempInterpreterFrame->GetCurrentLoopNum();
-                    Assert(loopNum != LoopHeader::NoLoop);
-                }
-                pCodeAddr = (DWORD_PTR)this->GetCurrentCodeAddr();
+                return offset;
             }
+#endif
+        }
+        return offset;
+    }
 
-            // If the current instruction's return address is the beginning of the next statement then we will show error for the next line, which would be completely wrong.
-            // The quick fix would be to look the address which is at least lesser than current return address.
+    bool JavascriptStackWalker::TryGetByteCodeOffsetFromInterpreterFrame(uint32& offset) const
+    {
+#if ENABLE_NATIVE_CODEGEN
+        if (this->lastInternalFrameInfo.codeAddress != nullptr)
+        {
+            return false;
+        }
+#endif
+        offset = this->interpreterFrame->GetReader()->GetCurrentOffset();
+        if (offset == 0)
```

Comment:
My intention was to keep the original comment about debugger and OOS for the case when offset is 0. Even by making the condition as offset!=0 I'd want to keep the comment